File: card_validator/CheckCard.java
public class CheckCard {
	
	public static boolean check (String string) {
		int strl = string.length();
		if (strl < 1 | strl > 20) return false;
		int a = Character.getNumericValue(string.charAt(0));
		if (a < 2 | a > 7) return false;
		String new_string = "";
		for (int i = 0; i < strl; i++) {
		new_string += string.charAt(strl-1-i);
		}
		int summ = 0;
		int nstrl = new_string.length();
		for (int j = 0; j < nstrl; j++) {
			int b = Character.getNumericValue(new_string.charAt(j))*2;
			if (j % 2 != 0) {
				if (b > 9) summ += b - 9;
				else summ += b;
			}
			else summ += Character.getNumericValue(new_string.charAt(j));
		}
		if (summ % 10 == 0) return true;
		else return false;
	}
}

File: card_validator/checkcard.py
def checkcard(string):
	if len(string) == 0 or len(string) > 20:
		return False
	if int(string[0]) < 2 or int(string[0]) > 7:
		return False
	else:
		new_string = ''
		for i in range(len(string)):
			new_string = new_string + string[len(string)-1-i]
		summ = 0
		for j in range (len(new_string)):
			if j % 2 != 0:
				if int(new_string[j])*2 > 9:
					summ += int(new_string[j])*2 - 9
				else:
					summ += int(new_string[j])*2
			else:
				summ += int(new_string[j])
		if summ % 10 == 0:
			return True
		else:
			return False
